src/vs/workbench/contrib/remote/electron-sandbox/remote.contribution.ts

```diff
@@ -244,3 +250,39 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 	metadata: { description: SaveLocalFileCommand.LABEL, args: [] },
 	handler: SaveLocalFileCommand.handler()
 });
+
+registerAction2(class extends Action2 {
+	constructor() {
+		super({
+			id: 'workbench.remote.action.closeUnusedPorts',
+			title: localize('remote.actions.closeUnusedPorts', 'Close Unused Forwarded Ports'),
+			category: localize('remote.category', 'Remote'),
+			menu: [{
+				id: MenuId.CommandPalette
+			}],
+			precondition: ContextKeyExpr.and(ContextKeyExpr.notEquals(`config.${PORT_AUTO_SOURCE_SETTING}`, PORT_AUTO_SOURCE_SETTING_OUTPUT), RemoteNameContext)
+		});
+	}
+
+	async run(accessor: ServicesAccessor) {
+		const remoteExplorerService = accessor.get(IRemoteExplorerService);
+		const ports: Tunnel[] = [];
+		// collect all forwarded ports and filter out those who do not have a process running
+		const forwarded = remoteExplorerService.tunnelModel.forwarded;
+		for (const [_, tunnel] of forwarded) {
+			if (!tunnel.hasRunningProcess) {
```

Comment:
I think if you check `tunnel.hasRunningProcess === false` here instead of checking for undefined then you'll only close tunnels for which we don't have a running process, not simple ones which we don't know about (like on Windows). Can you try it out and see if that's the case?

Reply:
That makes sense. It still worked for me after changing this.